tests/common/test_oresat_file_cache.py

```diff
@@ -0,0 +1,299 @@
+import sys
+sys.path.append('../../')
+
+import unittest
+
+import threading
+
+from time import sleep
+
+from shutil import rmtree
+
+from os import rmdir, remove, listdir
+
+from os.path import basename, abspath, exists
+
+from olaf.common.oresat_file_cache import OreSatFileCache
+
+from olaf.common.oresat_file import new_oresat_file
+
+
+class TestOreSatFileCache(unittest.TestCase):
+    def test_oresat_file_cache__init__(self):
+
+        # TEST:non-existing directory name
+        dir_name = 'delete_me'
+        result = OreSatFileCache(dir_name)
+        self.assertEqual(result._dir, abspath(dir_name) + '/')
+        self.assertEqual(result.dir, abspath(dir_name))
+        self.assertFalse(result._data)
+        self.assertTrue(result._lock)
+        self.assertEqual(OreSatFileCache.__len__(result), 0)
+        rmdir(abspath(dir_name))
+
+        # TEST: non-existing file name used as dir name
+        dir_name = 'delete_me.txt'
+        result = OreSatFileCache(dir_name)
+        self.assertEqual(result._dir, abspath(dir_name) + '/')
+        self.assertEqual(result.dir, abspath(dir_name))
+        self.assertFalse(result._data)
+        self.assertTrue(result._lock)
+        self.assertEqual(OreSatFileCache.__len__(result), 0)
+        rmdir(abspath(dir_name))
+
+        # TEST: existing directory name
+        dir_name = 'existing_dir'
+        OreSatFileCache(dir_name)
+        result = OreSatFileCache(dir_name)
+        self.assertEqual(result._dir, abspath(dir_name) + '/')
+        self.assertEqual(result.dir, abspath(dir_name))
+        self.assertFalse(result._data)
+        self.assertTrue(result._lock)
+        self.assertEqual(OreSatFileCache.__len__(result), 0)
+        rmdir(abspath(dir_name))
+
+        # TEST: existing file name used as dir name
+        dir_name = 'existing_file.xyz'
+        with open(dir_name, 'w'):
+            pass
```

Comment:
Could use `touch()` from the `pathlib` library instead

Reply:
Decided to stick to using the open() function instead of importing pathlib to use touch().